### PR TITLE
chore: use new eslint and jest configs

### LINKED ISSUE #12: Update eslint-config
Port this PR into this repository: https://github.com/moxystudio/next-with-moxy/pull/75

The config is slightly different as you should not enable the a11y web package. Be mindful that we should add a `eslint-config-react-native` enhancer to our eslint-config, and use it here.

### PR DESCRIPTION
**Pull request by satazor**
Closes #12

### CODE REVIEW

**Comment by satazor** [comment]
A few remarks:

- The `posinstall` script is not working on windows. If windows compatibility is not necessary, then I would say to remove `cross-env`. Also see https://shift.infinite.red/getting-started-with-react-native-development-on-windows-90d85a72ae65.
- Not sure if `@babel/core`  and `@babel/runtime` dev dependencies are needed?
- The eslint config is using `env` set to browser, but we should instead add a `eslint-config-react-native` enhancer and use it.

**Comment by acostalima** [comment]
> A few remarks:
> 
> * The `posinstall` script is not working on windows. If windows compatibility is not necessary, then I would say to remove `cross-env`. Also see [shift.infinite.red/getting-started-with-react-native-development-on-windows-90d85a72ae65](https://shift.infinite.red/getting-started-with-react-native-development-on-windows-90d85a72ae65).

Yep, the `pod` command is only used for iOS and we cannot build the iOS app on Windows. I know `pod` is emitting an error related to the build target. We shall address that when we have the App Store Connect setup ready. Until then, we're not able to build the iOS app, not even on MacOS.

> * Not sure if `@babel/core`  and `@babel/runtime` dev dependencies are needed?

Probably not. I think we can remove both, indeed.

> * The eslint config is using `env` set to browser, but we should instead add a `eslint-config-react-native` enhancer and use it.

👌

**Comment by satazor** [comment]
Alright, shall we remove the `cross-env` dependency and from scripts then?

**Comment by acostalima** [comment]
> Alright, shall we remove the `cross-env` dependency and from scripts then?

Yes, please! 🙏

**Comment by satazor** [comment]
@acostalima I've added another remark:

> There's a peer dependency warning on react because we are locking react to a specific old version. Is that necessary?

**Comment by satazor** [comment]
Also please test the boilerplate to see if the app is working correctly before merging.

**Comment by acostalima** [comment]
> @acostalima I've added another remark:
> 
> > There's a peer dependency warning on react because we are locking react to a specific old version. Is that necessary?

As far as my knowledge goes, not really. I think we update to the latest minor version.

**Comment by acostalima** [comment]
> > Alright, shall we remove the `cross-env` dependency and from scripts then?
> 
> Yes, please! 🙏

Well, actually, we should be able to build the Android app 😛

File: package.json [hunk 1]
```diff
@@ -52,28 +52,21 @@
...
     "eslint": "^6.8.0",
     "husky": "^4.2.0",
     "jest": "^25.1.0",
-    "jest-enzyme": "^7.0.2",
-    "jest-in-case": "^1.0.2",
-    "jsdom": "^16.2.1",
     "lint-staged": "^10.0.2",
     "metro-react-native-babel-preset": "^0.59.0",
-    "react-dom": "^16.8.3",
+    "react-dom": "^16.13.1",
```
Comment (acostalima) [inline package.json]: Could `react-dom` be a dependency of @moxy/jest-config only instead? 🤔

Comment (acostalima) [reply package.json]: Removed `react-dom` from our dependency list.

Comment (acostalima) [reply package.json]: I've just noticed enzyme-adapter-react-16 has a peer dependency on react-dom. 🙏

**Pull request merged**